### PR TITLE
Add the option to filter the ranking

### PR DESCRIPTION
**Pull request by helenatran**
This PR creates buttons which allow the user to filter the ranking to:
- view a ranking of both restaurants and cafes
- view a ranking of only restaurants
- view a ranking of only cafes

Screen Shot 2021-02-12 at 7 27 03 pm

### CODE REVIEW

File: src/main/webapp/src/app/ranking/ranking-list/ranking-list.component.ts [hunk 1]
```diff
@@ -26,6 +30,12 @@ export class RankingListComponent implements OnInit {
   ngDoCheck() {
     if (this.hiddenGems !== this.appComponent.hiddenGems) {
       this.hiddenGems = this.appComponent.hiddenGems;
```
Comment (lorikeet-s) [inline src/main/webapp/src/app/ranking/ranking-list/ranking-list.component.ts:32]: There's a lot more arrays now than there were before, it's obvious what's happening when you look at the html file but could you add a comment here or at the top of the class explaining briefly what the arrays are used for so flipping over the html file is not necessary

**Review by lorikeet-s — CHANGES_REQUESTED** [review]

**Review by lorikeet-s — APPROVED** [review]

File: src/main/webapp/src/app/ranking/ranking-list/ranking-list.component.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { Component, OnInit } from '@angular/core';
 import { AppComponent } from 'src/app/app.component';
 import { LocationService } from '../../location.service';
 import { Location } from 'src/app/models/location';
```
Comment (sehhajdkg) [inline src/main/webapp/src/app/ranking/ranking-list/ranking-list.component.ts:4]: nit - we can remove location and location service now that we have the app component hidden gems

File: src/main/webapp/src/app/ranking/ranking-list/ranking-list.component.ts [hunk 3]
```diff
@@ -10,10 +11,17 @@ import { Location } from 'src/app/models/location';
   styleUrls: ['./ranking-list.component.css']
 })
 export class RankingListComponent implements OnInit {
-
+  // There are 3 arrays of hidden gems to allow users to filter the ranking to
+  // see either a ranking of restaurants, a ranking of cafes, or a ranking of both
   hiddenGems = this.appComponent.hiddenGems;
+  restaurants = [] as HiddenGem[];
+  cafes = [] as HiddenGem[];
   location = {} as Location;
```
Comment (sehhajdkg) [inline src/main/webapp/src/app/ranking/ranking-list/ranking-list.component.ts:19]: delete hehe

File: src/main/webapp/src/app/ranking/ranking-list/ranking-list.component.ts [hunk 4]
```diff
@@ -10,10 +11,17 @@ import { Location } from 'src/app/models/location';
...
 
+  // This shows which ranking list will be displayed
+  // Options: only restaurants, only cafes, or both
+  show = "any";
+
   constructor(private locationService: LocationService, private appComponent: AppComponent) { }
```
Comment (sehhajdkg) [inline src/main/webapp/src/app/ranking/ranking-list/ranking-list.component.ts:25]: remove location service from here :)

**Review by sehhajdkg — APPROVED** [review]

**Pull request merged**